File: ticket-backend/src/main/java/aics/server/api/events/dtos/FetchEventsFilterOptionsDto.java
package aics.server.api.events.dtos;

import aics.domain.event.dtos.EventsFilterOptionsDto;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;

@Data
@Accessors(chain = true)
public class FetchEventsFilterOptionsDto implements Serializable {

    private EventsFilterOptionsDto options;
}

File: ticket-frontend/src/root/events/list/dtos/fetch-events-filter-options-dto.ts
import { EventsFilterOptionsDto } from '../../../../modules/event/dtos/events-filter-options-dto';

export class FetchEventsFilterOptionsDto {
    options: EventsFilterOptionsDto | null = null;

    static fromObj(obj: any): FetchEventsFilterOptionsDto | null {
        if (!obj) {
            return null;
        }
        const fetchEventsFilterOptionsDto: FetchEventsFilterOptionsDto = new FetchEventsFilterOptionsDto();
        fetchEventsFilterOptionsDto.options = EventsFilterOptionsDto.fromObj(obj.options);
        return fetchEventsFilterOptionsDto;
    }
}